scripts/xcm/setup.js

```diff
@@ -0,0 +1,273 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+require('dotenv').config();
+
+const {ApiPromise, WsProvider, Keyring} = require('@polkadot/api');
+const {cryptoWaitReady} = require('@polkadot/util-crypto');
+const {
+    relayChainProvider,
+    assetHubProvider,
+    bridgeHubProvider,
+    transferBalance,
+    setFeeHandler,
+    setMpcAddress,
+    registerDomain,
+    setFee,
+    setFeeRate,
+    getNativeAssetId,
+    createAsset,
+    setAssetMetadata,
+    mintAsset,
+    getUSDCAssetId,
+    getAHNMultiAsset,
+    getAHNAssetId,
+    getTTTMultiAsset,
+    getTTTAssetId,
+    queryBridgePauseStatus,
+    FeeReserveAccount,
+    NativeTokenTransferReserveAccount,
+    OtherTokenTransferReserveAccount,
+    hrmpChannelRequest,
+    getHRMPChannelMessage,
+    getHRMPChannelDest,
+    delay,
+    usdcAssetID,
+    usdcMinBalance,
+    usdcName,
+    usdcSymbol,
+    usdcDecimal,
+    ahnAssetID,
+    ahnMinBalance,
+    ahnName,
+    ahnSymbol,
+    ahnDecimal,
+    tttAssetID,
+    tttMinBalance,
+    tttName,
+    tttSymbol,
+    tttDecimal,
+    bhnAssetID,
+    bhnMinBalance,
+    bhnName,
+    bhnSymbol,
+    bhnDecimal,
+} = require("./util");
+
+const BN = require('bn.js');
+const bn1e12 = new BN(10).pow(new BN(12));
+const bn1e18 = new BN(10).pow(new BN(18));
+const bn1e20 = new BN(10).pow(new BN(20));
+
+const feeHandlerType = {
+    BasicFeeHandler: "BasicFeeHandler",
+    PercentageFeeHandler: "PercentageFeeHandler",
+    DynamicFeeHandler: "DynamicFeeHandler"
+}
+
+const supportedDestDomains = [
+    {
+        domainID: 1,
+        chainID: 1
+    }
+]
+
+async function main() {
+    // relay chain
+    const relayChainApi = await ApiPromise.create({
+        provider: relayChainProvider,
+    });
+    // asset hub parachain
+    const assetHubApi = await ApiPromise.create({
+        provider: assetHubProvider,
+    });
+    // bridge hub parachain
+    const bridgeHubApi = await ApiPromise.create({
+        provider: bridgeHubProvider,
+    });
+
+    // prepare keyring
+    await cryptoWaitReady();
+    const keyring = new Keyring({type: 'sr25519'});
+    const sudo = keyring.addFromUri('//Alice');
+
+    console.log('======= Relaychain setup begin =======');
+
+    // relaychain setup
+    // sovereignaccount of parachain 1000 on relaychain:
+    const sovereignAccount1000 = "5Ec4AhPZk8STuex8Wsi9TwDtJQxKqzPJRCH7348Xtcs9vZLJ";
+    // sovereignaccount of parachain 1013 on relaychain:
+    const sovereignAccount1013 = "5Ec4AhPcMD9pfD1dC3vbyKXoZdZjigWthS9nEwGqaSJksLJv";
```

Comment:
Instead of hardcode, you also can derive the account like [this way](https://substrate.stackexchange.com/a/9200/2246)

Reply:
thanks for sharing this man! def will be useful in rust, what I used is [this tool](https://www.shawntabrizi.com/substrate-js-utilities/) which is very convenient in js script